CommonApplication.cmake

```diff
@@ -122,6 +123,34 @@ function(common_application Name)
     endif()
   endif()
 
+  if(NOT THIS_GUI OR THIS_DOXYGEN)
+    # run binary with --help to capture output for doxygen
+    set(_doc "${PROJECT_BINARY_DIR}/help/${Name}.md")
+    set(_cmake "${CMAKE_CURRENT_BINARY_DIR}/${Name}.cmake")
+    file(WRITE ${_cmake} "
+      execute_process(COMMAND \${APP} --help
+      OUTPUT_FILE ${_doc} ERROR_FILE ${_doc}.error)
+      file(READ ${_doc} _doc_content)
+      if(NOT _doc_content)
+        message(FATAL_ERROR \"${Name} is missing --help\")
+      endif()
+      file(WRITE ${_doc} \"${Name} {#${Name}}
+============
+
+```
+\${_doc_content}
+```
+\")
+")
+    add_custom_command(OUTPUT ${_doc}
+      COMMAND ${CMAKE_COMMAND} -DAPP=$<TARGET_FILE:${Name}> -P ${_cmake}
+      DEPENDS ${Name}
+      COMMENT "Creating help for ${Name}")
+    add_custom_target(${Name}-help ALL DEPENDS ${_doc})
+    add_dependencies(${PROJECT_NAME}-all ${Name}-help)
```

Comment:
In my understanding:
- the "all" target is the default target when one does just "make", and it should just *build* everything (not install or do anything fancy).
- the project-all target is the equivalent of the "all" target for each subproject, in the same way that project-install is the equivalent of the global "install" target.
- the project-doxygen is our own concept and can do whatever it likes to produce documentation.
If the project-doxygen is not available at this point, then the solution is to reverse the dependency order, exactly like it is done with the ${PROJECT_NAME}-coverage target. Which is also run by project-doxygen and not by project-all...